backport.py

```diff
@@ -116,19 +116,26 @@ def __init__(self, token, organ_name, repo_name, debug=False):
         self.user_name = self.g.get_user().login
         self.debug = debug
 
-    def run(self, pr_num, target_branch, is_continue, abort_before_push):
-        assert isinstance(pr_num, int)
-        assert pr_num >= 1
+    def run(self, pr_num, sha, target_branch, is_continue, abort_before_push):
+        assert isinstance(pr_num, int) and pr_num >= 1 or pr_num is None
+        assert (pr_num is None and sha is not None) or (
+            pr_num is not None and sha is None
+        )
         assert isinstance(target_branch, str)
         assert isinstance(is_continue, bool)
         assert isinstance(abort_before_push, bool)
 
         # Get information of the original pull request
+        if sha is not None:
+            pr_num, branch_name, _ = self.parse_log_message(sha)
+
         pr = self.repo.get_pull(pr_num)
         if not pr.merged:
             raise GracefulError('PR #{} is not merged'.format(pr_num))
         merge_commit_sha = pr.merge_commit_sha
         _, branch_name, _ = self.parse_log_message(merge_commit_sha)
+        pr = self.repo.get_pull(pr_num)
+
```

Comment:
This seems unnecessarily repeated.

Reply:
yes, it was a mistake with yanking, thanks for pointing it out 🙇‍♂️